python/composio/tools/toolset.py

```diff
@@ -837,6 +785,153 @@ def _merge_processors(self, processors: ProcessorsType) -> None:
+                params[key] = self._load_file_as_param(
+                    _path=params[key],
+                    _accept=schema["properties"][key].get("accept", ["*/*"]),
+                )
+                continue
+
+            if schema["properties"][key]["type"] == "object":
+                params[key] = self._substitute_file_upload_data_recursively(
+                    params=params[key],
+                    schema=schema["properties"][key],
+                )
+
+        return params
+
+    def _substitute_file_upload_data(self, action: Action, params: t.Dict) -> t.Dict:
+        _schema = self._action_schemas.get(action.slug)
+        if _schema is None:
+            (_schema,) = self.get_action_schemas(actions=[action])
+
+        if _schema.parameters.file_uploadable:
+            return self._load_file_as_param(
+                _path=params["file"],
+                _accept=_schema.parameters.accept,
+            )
+
+        return self._substitute_file_upload_data_recursively(
+            params=params,
+            schema=_schema.model_dump().pop("parameters"),
+        )
+
+    def _is_file_downloadable(self, schema: t.Dict) -> bool:
+        if schema.get("title") == "FileDownloadable":
+            return True
+        if "properties" not in schema:
+            return False
+        return "name" in schema["properties"] and "content" in schema["properties"]
+
+    def _store_response_file(self, action: str, response: t.Dict) -> str:
+        outdir = self.output_dir / action
+        outdir.mkdir(parents=True, exist_ok=True)
+        outfile = outdir / response["name"]
+
+        try:
+            outfile.write_bytes(base64.b64decode(response["content"]))
+        except binascii.Error:
+            outfile.write_text(response["content"])
+        return str(outfile)
+
+    def _substitute_file_download_data_recursively(
+        self,
+        schema: t.Dict,
+        response: t.Dict,
+        action: Action,
+    ) -> t.Dict:
+        if "properties" not in schema:
+            return response
+
+        for key in schema["properties"]:
+            if self._is_file_downloadable(schema=schema["properties"][key]):
+                response[key] = self._store_response_file(
+                    action=action.slug,
+                    response=response[key],
+                )
+                continue
+
+            if "type" not in schema["properties"][key]:
+                continue
+
+            if schema["properties"][key]["type"] == "object":
+                response[key] = self._substitute_file_download_data_recursively(
+                    schema=schema["properties"][key],
+                    response=response[key],
+                    action=action,
+                )
+
+        return response
+
+    def _substitute_file_download_data(
+        self,
+        action: Action,
+        response: t.Dict,
+    ) -> t.Dict:
+        if not response["successfull"]:
```

Comment:
I think we are passing successful also now. Should we use that now?
cc: @utkarsh-dixit